File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/non_secrets/DeleteRecordTest.java
package org.hyperledger.indy.sdk.non_secrets;

import org.hyperledger.indy.sdk.wallet.WalletItemNotFoundException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;


public class DeleteRecordTest extends NonSecretsIntegrationTest {

	@Test
	public void testDeleteRecordWorks() throws Exception {
		WalletRecord.add(wallet, type, id, value, tags).get();
		WalletRecord.delete(wallet, type, id).get();

		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletItemNotFoundException.class));

		WalletRecord.get(wallet, type, id, optionsEmpty).get();
	}

	@Test
	public void testDeleteRecordWorksForTwice() throws Exception {
		WalletRecord.add(wallet, type, id, value, tags).get();

		WalletRecord.delete(wallet, type, id).get();

		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletItemNotFoundException.class));

		WalletRecord.delete(wallet, type, id).get();
	}

	@Test
	public void testDeleteRecordWorksForNotFoundRecord() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletItemNotFoundException.class));

		WalletRecord.deleteTags(wallet, type, id, "[\"tagName1\"]").get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/NonSecretsTests/DeleteRecordTest.cs
using Hyperledger.Indy.NonSecretsApi;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.NonSecretsTests
{
    [TestClass]
    public class DeleteRecordTest : NonSecretsIntegrationTestBase
    {
        [TestMethod]
        public async Task TestDeleteRecordWorks()
        {
            await NonSecrets.AddRecordAsync(wallet, type, id, value, tags);
            await NonSecrets.DeleteRecordAsync(wallet, type, id);

            var ex = await Assert.ThrowsExceptionAsync<WalletItemNotFoundException>(() =>
                NonSecrets.GetRecordAsync(wallet, type, id, optionsEmpty)
            );
        }

        [TestMethod]
        public async Task TestDeleteRecordWorksForTwice()
        {
            await NonSecrets.AddRecordAsync(wallet, type, id, value, tags);
            await NonSecrets.DeleteRecordAsync(wallet, type, id);

            var ex = await Assert.ThrowsExceptionAsync<WalletItemNotFoundException>(() =>
                NonSecrets.DeleteRecordAsync(wallet, type, id)
            );
        }

        [TestMethod]
        public async Task TestDeleteRecordWorksForNotFoundRecord()
        {
            var ex = await Assert.ThrowsExceptionAsync<WalletItemNotFoundException>(() =>
               NonSecrets.DeleteRecordAsync(wallet, type, id)
           );
        }
    }
}
